### PR TITLE
Parquet to bypass Arrow

### PR DESCRIPTION
**Pull request by whitphx**

### CODE REVIEW

File: lib/streamlit/type_util.py [hunk 1]
```diff
@@ -744,26 +745,46 @@ def fix_arrow_incompatible_column_types(
...
-    except (pa.ArrowTypeError, pa.ArrowInvalid, pa.ArrowNotImplementedError) as ex:
+        df.to_parquet(buf)
+    except ValueError as ex:
         _LOGGER.info(
-            "Serialization of dataframe to Arrow table was unsuccessful due to: %s. "
+            "Serialization of dataframe to Parquet table was unsuccessful due to: %s. "
             "Applying automatic fixes for column types to make the dataframe Arrow-compatible.",
             ex,
         )
         df = fix_arrow_incompatible_column_types(df)
```
Comment (lukasmasuch) [inline lib/streamlit/type_util.py:810]: just a note: parquet might have other incompatibilities compared to the arrow serialization, which might be good to add here once we have identified some column types that don't work.

**Review by lukasmasuch — APPROVED** [review]
LGTM 👍  This is awesome! It will unlock a lot of interesting features we released in the last couple of months 🙌  

However there might be some incompatibilities with some of the less commonly used column types. You could potentially test this out by running some of the scripts we use for e2e testing, e.g.: [st_data_editor_column_types.py](https://github.com/streamlit/streamlit/blob/develop/e2e/scripts/st_data_editor_column_types.py) or most of the `st_arrow_...` scripts

**Comment by lukasmasuch** [comment]
I did some tests, and most of the common data/column types work fine. This includes: string, boolean, integer, float, datetime 👍  This will already cover most use cases. Unfortunately, some of the less common data types don't work: lists, date, time, interval, period. We might be able to get them working with a bit more debugging, or - as a fallback - add them to `fix_arrow_incompatible_column_types` in stlite to at least be able to view the data instead of getting an exception.

**Comment by whitphx** [comment]
@LukasMasuch Thank you very much for such huge helps!

**Pull request merged**